### PR TITLE
Add eth2 github action

### PR DESCRIPTION
**Pull request by Spacesai1or**
## Description

Adds Github action that runs tests for ETH2 packages: `web3-eth2-core` and `web3-eth2-beaconchain`

Includes changes from #3890 so that ETH2 tests are passing

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

File: packages/web3-eth2-beaconchain/test/schemaMethods.test.ts [hunk 1]
```diff
@@ -135,7 +135,7 @@ it('getValidatorBalances', async () => {
     expect(response[0]).toMatchObject(expectedResponse)
 })
 
-it('getEpochCommittees', async () => {
+xit('getEpochCommittees', async () => {
```
Comment (Spacesai1or) [inline packages/web3-eth2-beaconchain/test/schemaMethods.test.ts:138]: Temp fix, getting a 400 because route is wrong, but I don't understand why I would've made the route what it is, if it wasn't supposed to be - need to further investigate and changes should probably be in a separate PR

File: packages/web3-eth2-beaconchain/package.json [hunk 2]
```diff
@@ -10,18 +10,27 @@
     "types": "types.ts",
     "scripts": {
         "test": "jest",
-        "tsc": "babel src --out-dir lib --extensions \".ts,.tsx\" --source-maps inline"
+        "tsc": "babel src --out-dir lib --extensions \".ts,.tsx\" --source-maps inline",
```
Comment (GregTheGreek) [inline packages/web3-eth2-beaconchain/package.json:13]: should this be renamed ?

File: packages/web3-eth2-beaconchain/test/schemaMethods.test.ts (hunk 1 above)
Comment (GregTheGreek) [inline packages/web3-eth2-beaconchain/test/schemaMethods.test.ts:138]: typo?

Comment (Spacesai1or) [reply packages/web3-eth2-beaconchain/test/schemaMethods.test.ts:138]: Sorry [this](https://github.com/ChainSafe/web3.js/pull/3892#discussion_r567209601) got posted, but it was still pending

Comment (Spacesai1or) [reply packages/web3-eth2-beaconchain/test/schemaMethods.test.ts:138]: Related [issue](https://github.com/ChainSafe/web3.js/issues/3893)

File: packages/web3-eth2-beaconchain/package.json (hunk 2 above)
Comment (Spacesai1or) [reply packages/web3-eth2-beaconchain/package.json:13]: I would like it to be, but some of our CI scripts use `lerna exec tsc`. Perhaps we can open a separate PR to update the scripts/npm commands to `compile` instead of `tsc`?

Comment (GregTheGreek) [reply packages/web3-eth2-beaconchain/package.json:13]: yeah that makes sense!

**Review by GregTheGreek — APPROVED** [review]
